spark-extension/src/main/scala/org/apache/spark/sql/blaze/NativeConverters.scala

```diff
@@ -843,13 +843,17 @@ object NativeConverters extends Logging {
           }
         }
       case e: Ceil if !e.dataType.isInstanceOf[DecimalType] =>
-        buildExprNode {
-          _.setTryCast(
-            pb.PhysicalTryCastNode
-              .newBuilder()
-              .setExpr(buildScalarFunction(pb.ScalarFunction.Ceil, e.children, e.dataType))
-              .setArrowType(convertDataType(e.dataType))
-              .build())
+        if (e.child.dataType.isInstanceOf[LongType]) {
```

Comment:
can we just change the condition to `case e: Ceil if !e.dataType.isInstanceOf[DecimalType] && e.child.dataType.isInstanceOf[LongType] => ...`

Reply:
Native ceil function will be no proceess when the data type is `LongType`, it will help to avoid falling back to spark.

`
25/02/16 10:32:34 WARN NativeConverters: Falling back expression: scala.NotImplementedError: unsupported expression: (class org.apache.spark.sql.catalyst.expressions.Ceil) CEIL(cast(id#9 as bigint))
`